pkgs/desktops/deepin/deepin-terminal/default.nix

```diff
@@ -1,55 +1,61 @@
-{ stdenv, fetchurl, fetchFromGitHub, pkgconfig, gtk3, vala, cmake,
-  ninja, vte, libgee, wnck, zssh, gettext, librsvg, libsecret,
-  json-glib, gobjectIntrospection, deepin-menu, deepin-shortcut-viewer
-}:
+{ stdenv, fetchurl, fetchFromGitHub, pkgconfig, cmake, ninja, vala,
+  gettext, gobjectIntrospection, at-spi2-core, dbus, epoxy, expect,
+  gtk3, json-glib, libXdmcp, libgee, libpthreadstubs, librsvg,
+  libsecret, libtasn1, libxcb, libxkbcommon, p11-kit, pcre, vte, wnck,
+  deepin-menu, deepin-shortcut-viewer, deepin }:
 
 stdenv.mkDerivation rec {
-  name = "deepin-terminal-${version}";
-  version = "3.0.3";
+  name = "${pname}-${version}";
+  pname = "deepin-terminal";
+  version = "3.0.8";
 
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = "deepin-terminal";
     rev = version;
-    sha256 = "04yvim97a4j8fq5lq2g6svs8qs79np9m4nl6x83iv02wkb9b7gqa";
+    sha256 = "0f2dbs3nfpxwar0zn029gv4kpzdg47189fijvdpmkvkdky1b5smj";
   };
 
-  patches = [
-    # Do not build vendored zssh and vte
-    (fetchurl {
-      name = "remove-vendor.patch";
-      url = https://git.archlinux.org/svntogit/community.git/plain/trunk/remove-vendor.patch?h=packages/deepin-terminal&id=de701614c19c273b98b60fd6790795ff7d8a157e;
-      sha256 = "0g7hhvr7ay9g0cgc6qqvzhbcwvbzvrrilbn8w46ypfzj7w5hlkqv";
-    })
-  ];
-
-  postPatch = ''
-    substituteInPlace ssh_login.sh --replace /usr/lib/deepin-terminal/zssh "${zssh}/bin/zssh"
```

Comment:
Is zssh no longer used? We can remove it from nixpkgs since it was only added for deepin-terminal.

Reply:
`zssh` is still used, but its path is [not hard-coded in `deepin-terminal` anymore](https://github.com/linuxdeepin/deepin-terminal/blob/master/ssh_login.sh#L40). It is supposed to be in `$PATH`.  It provides Remote Management support.

In archlinux `zssh` is an _optional_ dependency, In Fedora it is a _recommended_ dependency. In debian it is normal dependency.

So I think `zssh` should not be removed from nixpkgs.